lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -606,6 +606,31 @@ void BlocksReachingTensorCode::dump() {
 //                             FunctionPartitioner
 //===----------------------------------------------------------------------===//
 
+namespace llvm {
+template<typename T> struct DenseMapInfo;
+
+// An alternative is to SourceManager::getLineAndColumn() as the
+// DenseMap/DenseSet key type instead of SourceRange. That would require that
+// the call-site translate a SILLocation / SourceRange to line and column
+// numbers first.
+template<> struct DenseMapInfo<SourceRange> {
```

Comment:
This should go in the header that defines SourceRange, and upstreamed :-)

Reply:
Will do!